mrgeo-mapalgebra/mrgeo-mapalgebra-image/src/main/scala/org/mrgeo/mapalgebra/IngestImageMapOp.scala

```diff
@@ -104,21 +105,59 @@ class IngestImageMapOp extends RasterMapOp with Externalizable {
 
   override def execute(context: SparkContext): Boolean = {
 
-    val inputs = input.getOrElse(throw new IOException("Inputs not set"))
+    val inputfile = input.getOrElse(throw new IOException("Inputs not set"))
 
-    val path = new Path(inputs)
-    val fs = HadoopFileUtils.getFileSystem(context.hadoopConfiguration, path)
+    val filebuilder = Array.newBuilder[String]
 
-    val rawfiles = fs.listFiles(path, true)
+    var f: File = null
+    try {
+      f = new File(new URI(inputfile))
+    }
+    catch {
+      case ignored: Any => f = new File(inputfile)
+    }
 
-    val filebuilder = Array.newBuilder[String]
-    while (rawfiles.hasNext) {
-      val raw = rawfiles.next()
+    def walk(dir:File):Array[String] = {
+      val files = Array.newBuilder[String]
+      val dir: Array[File] = f.listFiles
+      if (dir != null) {
+        for (s <- dir) {
+          try {
+            if (s.isFile) {
+              files += s.toURI.toString
+            }
+            else if (s.isDirectory) {
+              files ++= walk(s)
+            }
+          }
+        }
+      }
+      files.result()
+    }
 
-      if (!raw.isDirectory) {
-        filebuilder += raw.getPath.toUri.toString
+    if (f.exists()) {
+      if (f.isFile) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/DGIS/mrgeo/file/2578647041/issues/source?bid=3293399&fileBranchId=3301986#l139)